src/control/server/mgmt_system.go

```diff
@@ -71,6 +71,15 @@ func (svc *mgmtSvc) GetAttachInfo(ctx context.Context, req *mgmtpb.GetAttachInfo
 		}
 	}
 
+	getPrimaryProvider := func() string {
+		return svc.clientNetworkHint[0].Provider
+	}
+
+	getSecondaryProvider := func(idx int) string {
+		// Primary is at idx 0, secondary providers start afterward
+		return svc.clientNetworkHint[idx+1].Provider
+	}
```

Comment:
Should this helper guard against `idx > len(svc.clientNetworkHint) - 1` ?

Reply:
It can't actually happen in the usage, since we check the length of the clientNetHint array against each set of secondary URIs below. Since the scope is limited to this function, I didn't think guarding was necessary.

Reply:
OK, fine with me. Just wanted to make sure.